numpy/core/src/multiarray/descriptor.c

```diff
@@ -734,22 +729,12 @@ _convert_from_commastring(PyObject *obj, int align)
         return NULL;
     }
     if (PyList_GET_SIZE(listobj) == 1) {
-        int retcode;
-        retcode = PyArray_DescrConverter(PyList_GET_ITEM(listobj, 0),
-                                                &res);
-        if (retcode == NPY_FAIL) {
-            res = NULL;
-        }
+        res = _convert_from_any(PyList_GET_ITEM(listobj, 0), align);
```

Comment:
This fixes an extension of the bug in similar nonsensical code to https://github.com/numpy/numpy/pull/15287#issue-360237537, namely
```python
np.typeDict['mypoint'] = [('a', 'i1'), ('b', 'i2')]
assert np.dtype('mypoint,', align=True).isalignedstruct
```